spec/services/resync_service_spec.rb

```diff
@@ -0,0 +1,142 @@
+          unpublish_params = hash_including(
+            type: "withdrawal",
+            locale: document.current_edition.locale,
+          )
+          expect(GdsApi.publishing_api_v2).to receive(:put_content).once.ordered
+          expect(GdsApi.publishing_api_v2).to receive(:unpublish).once.
+            with(document.content_id, unpublish_params).ordered
+          ResyncService.call(document)
+        end
+      end
+
+      context "when the live edition is unpublished with redirect" do
+        before do
+          document.current_edition.stub(:removed?).and_return(true)
+          document.current_edition.status.details.stub(:redirect?).and_return(true)
+          document.current_edition.status.details.stub(:alternative_path).and_return("/foo/bar")
+          document.current_edition.status.details.stub(:explanatory_note).and_return("Explanation")
+        end
+
+        it "unpublishes the edition as redirected" do
+          unpublish_params = hash_including(
+            type: "redirect",
+            locale: document.current_edition.locale,
+            alternative_path: "/foo/bar",
+            explanation: "Explanation",
+          )
+          expect(GdsApi.publishing_api_v2).to receive(:put_content).once.ordered
+          expect(GdsApi.publishing_api_v2).to receive(:unpublish).once.
+            with(document.content_id, unpublish_params).ordered
+          ResyncService.call(document)
+        end
+      end
+
+      context "when the live edition is unpublished without a redirect" do
+        before do
+          document.current_edition.stub(:removed?).and_return(true)
+          document.current_edition.status.details.stub(:redirect?).and_return(false)
+          document.current_edition.status.details.stub(:alternative_path).and_return("")
+          document.current_edition.status.details.stub(:explanatory_note).and_return("")
+        end
+
+        it "unpublishes the edition as redirected" do
+          unpublish_params = hash_including(
+            type: "gone",
+            locale: document.current_edition.locale,
+            alternative_path: "",
+            explanation: "",
+          )
+          expect(GdsApi.publishing_api_v2).to receive(:put_content).once.ordered
+          expect(GdsApi.publishing_api_v2).to receive(:unpublish).once.
+            with(document.content_id, unpublish_params).ordered
+          ResyncService.call(document)
+        end
+      end
+    end
+
+    context "when there are both live and current editions" do
+      let(:document) { create(:document, :with_current_and_live_editions) }
+
+      it "updates the system_political value associated with both editions" do
+        expect(document.live_edition.system_political).to be false
+        expect(document.current_edition.system_political).to be false
+        ResyncService.call(document)
+        expect(document.live_edition.system_political).to be true
+        expect(document.current_edition.system_political).to be true
```

Comment:
Might want to refactor this as: 

```
expect { ResyncService.call(document) }
  .to change { document.live_edition.system_political }.to(true)
  .and change { document.current_edition.system_political }.to(true)
```

You also might want to mock the PoliticalEditionIdentifier to return true otherwise I'm not really sure what about this document is supposed to mean that this content is political.